realm-transformer/src/main/kotlin/io/realm/transformer/build/IncrementalBuild.kt

```diff
@@ -0,0 +1,155 @@
+import javassist.NotFoundException
+import org.gradle.api.Project
+import java.io.File
+import java.util.jar.JarFile
+
+class IncrementalBuild(project: Project, outputProvider: TransformOutputProvider, transform: RealmTransformer)
+    : BuildTemplate(project, outputProvider, transform) {
+
+    override fun prepareOutputClasses(inputs: MutableCollection<TransformInput>) {
+        this.inputs = inputs;
+        findClassNames(inputs, outputClassNames, outputReferencedClassNames) // Output files
+        logger.debug("Incremental build. Files being processed: ${outputClassNames.size}.")
+        logger.debug("Incremental files: ${outputClassNames.joinToString(",")}")
+    }
+
+    override fun filterForModelClasses(outputClassNames: Set<String>, outputReferencedClassNames: Set<String>) {
+        outputModelClasses.addAll(findModelClasses(outputClassNames))
+    }
+
+    override fun transformDirectAccessToModelFields() {
+        // Use accessors instead of direct field access
+        outputClassNames.forEach {
+            logger.debug("Modify accessors in class: $it")
+            val ctClass: CtClass = classPool.getCtClass(it)
+            BytecodeModifier.useRealmAccessors(classPool, ctClass, null)
+            ctClass.writeFile(getOutputFile(outputProvider).canonicalPath)
+        }
+
+    }
+
+
+    /**
+     * Go through the transform input in order to find all files we need to transform.
+     *
+     * @param inputs set of input files
+     * @param directoryFiles the set of files in directories getting compiled. These are candidates for the transformer.
+     * @param referencedFiles the set of files that are possible referenced but never transformed (required by JavaAssist).
+     * @param isIncremental `true` if build is incremental.
+     */
+    override fun findClassNames(inputs: Collection<TransformInput>,
+                                directoryFiles: MutableSet<String>,
+                                referencedFiles: MutableSet<String>) {
+        inputs.forEach {
+            // Files in directories are files we most likely want to transform
+             it.directoryInputs.forEach {
+                val dirPath: String = it.file.absolutePath
+
+                 it.changedFiles.entries.forEach dirFile@ {
+                    if (it.value == Status.NOTCHANGED || it.value == Status.REMOVED) {
+                        return@dirFile
+                    }
+                    val filePath: String = it.key.absolutePath
+                    if (filePath.endsWith(SdkConstants.DOT_CLASS)) {
+                        val className = filePath
+                                .substring(dirPath.length + 1, filePath.length - SdkConstants.DOT_CLASS.length)
+                                .replace(File.separatorChar, '.')
+                        directoryFiles.add(className)
+                    }
+                }
+            }
+
+            // Files in Jars are always treated as referenced input. They should already have been
```

Comment:
This is contradicting https://github.com/realm/realm-java/pull/5925/files#diff-a0391da0b278a5ddb5a45b7866e136a8R61 description
The method is supposed to find all files we need to transform, however, the files in the Jars were already transformed?